lib/service_actor/collectionable.rb

```diff
@@ -8,26 +8,72 @@
 #   class Pay < Actor
 #     input :provider, inclusion: ["MANGOPAY", "PayPal", "Stripe"]
 #   end
+#
+#   class Pay < Actor
+#     input :provider,
+#           inclusion: {
+#             in: ["MANGOPAY", "PayPal", "Stripe"],
+#             message: (lambda do |_input_key, value, _inclusion_in|
+#               "Payment system \"#{value}\" is not supported"
+#             end)
+#           }
+#   end
 module ServiceActor::Collectionable
   def self.included(base)
     base.prepend(PrependedMethods)
   end
 
   module PrependedMethods
-    def _call
+    DEFAULT_MESSAGE = lambda do |input_key, value, inclusion_in|
+      "Input #{input_key} must be included " \
+      "in #{inclusion_in.inspect} but instead " \
+      "was #{value.inspect}"
+    end
+
+    private_constant :DEFAULT_MESSAGE
+
+    def _call # rubocop:disable Metrics/MethodLength
       self.class.inputs.each do |key, options|
+        value = result[key]
+
         # DEPRECATED: `in` is deprecated in favor of `inclusion`.
         inclusion = options[:inclusion] || options[:in]
-        next unless inclusion
 
-        next if inclusion.include?(result[key])
+        base_arguments = {
+          input_key: key,
+          value: value
+        }
+
+        inclusion_in, message = define_inclusion_from(inclusion)
 
-        raise ServiceActor::ArgumentError,
-              "Input #{key} must be included in #{inclusion.inspect} " \
-              "but instead was #{result[key].inspect}"
+        next if inclusion_in.nil?
+        next if inclusion_in.include?(value)
+
+        raise_error_with(
+          message,
+          **base_arguments,
+          inclusion_in: inclusion_in,
+        )
       end
 
       super
     end
+
+    private
+
+    def define_inclusion_from(inclusion)
+      message = DEFAULT_MESSAGE
+
+      if inclusion.is_a?(Hash) # advanced mode
+        inclusion_in, message = inclusion.values_at(:in, :message)
+      else
+        inclusion_in = inclusion
+      end
+
+      [
+        inclusion_in,
+        message
+      ]
```

Comment:
We could simplify this method down to:

```suggestion
      if inclusion.is_a?(Hash)
        inclusion.values_at(:in, :message)
      else
        [inclusion, DEFAULT_MESSAGE]
      end
```